### ISSUE TITLE
prism-line-highlight is slow on resize because it forces a reflow for every

### ISSUE THREAD

**Issue by justinbmeyer** [PrismJS/prism#1860]
**Information:**
- Prism version: `1.16.0`.
- Environment: Chrome Browser

Does the latest version of Prism from the [download page](https://prismjs.com/download.html) also have this issue? I'm using it from `npm`.

**Description**

The line-highlight plugin is causing multiple forced reflows every time I resize a page.  I have about 29 highlighted code blocks on the page. You can see this and the warning here:

![image](https://user-images.githubusercontent.com/78602/55927725-b2f6e080-5bdb-11e9-8895-a02b281932f0.png)

Looking at the code, there is a lot of repeated __read layout and write layout__ loops around here:

https://github.com/PrismJS/prism/blob/d2089d83509c1bcc3a5e74cbfc896680b2f76560/plugins/line-highlight/prism-line-highlight.js#L70

This is in a for-loop for each highlighted range, which is in another for loop for each code block.

Could this be changed to read the layout all at once and write the layout all at once?  If interested, __I will submit a PR for this__.  Here's [an example](https://www.bitovi.com/blog/canjs-4.0#queues) that shows the difference (500ms to 5ms) if batched read/writes are used.

Thanks!

**Comment by Golmote** [comment]
Hello! Thanks for opening this issue. If you're willing to submit a PR to fix this, please do, it is much appreciated!

**Comment by justinbmeyer** [comment]
[Here's the code](https://gist.github.com/justinbmeyer/1230ae64918372ea9bf124f6dce74326) (a PR will be created tomorrowish).  It changes my highlight time from about 400ms to about 2ms on the slowest performing page.

In short, I added a little helper that allows me to store calling a function or setting a property:

```js
function makeActions() {
	var actions = [];

	return {
		actions: actions,
		apply: function(fn, context, args){
			actions.push(function(){
				fn.apply(context, args);
			});
		},
		set: function(obj, key, value) {
			actions.push(function(){
				obj[key] = value;
			});
		}
	};
}
```

Then, instead of mutating the DOM, I use the helper:

```js
// line.setAttribute('data-range', currentRange);
action.apply(line.setAttribute, line,['data-range', currentRange]);

// line.style.top = startNode.offsetTop + 'px';
action.set(line.style, "top",  startNode.offsetTop + 'px');
```

And then return these actions to be called later:

```js
		var actions = [];
		Array.prototype.forEach.call(preElements, function (pre) {
			actions.push.apply(actions, highlightLines(pre) );
		});
		actions.forEach(function(action){
			action();
		});
```

**Comment by RunDevelopment** [comment]
Resolved by #1865.
